app/models/mixins/authentication_mixin.rb

```diff
@@ -386,6 +387,19 @@ def raw_change_password(_current_password, _new_password)
     raise NotImplementedError, _("must be implemented in subclass.")
   end
 
+  def assign_nested_endpoints(attributes)
+    record = endpoints.where(:role => attributes['role']).first_or_initialize
+    record.assign_attributes(attributes)
+    record # `assign_attributes` always returns `nil`
+  end
+
+  def assign_nested_authentications(attributes)
+    record = authentications.where(:authtype => attributes['authtype']).first_or_initialize
+    klass = authentication_class(attributes['authtype'])
```

Comment:
Won't this always just return `AuthUseridPassword` regardless (I don't see any changes in subclasses or models that this is mixed into)

Reply:
Well, I have to do them in provider repos :laughing:

Reply:
Here's the one for openstack: https://github.com/ManageIQ/manageiq-providers-openstack/pull/568/files#diff-6f9b687ce51fd31aa102fa9305ef2501R549